sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpAnnotatedMessageExtensions.cs

```diff
@@ -0,0 +1,408 @@
+           }
+
+           if (instance.HasSection(AmqpMessageSection.Properties))
+           {
+               CopyPropertiesSection(instance.Properties, clone.Properties);
+           }
+
+           if (instance.HasSection(AmqpMessageSection.Footer))
+           {
+               CopyDictionary(instance.Footer, clone.Footer);
+           }
+
+           if (instance.HasSection(AmqpMessageSection.DeliveryAnnotations))
+           {
+               CopyDictionary(instance.DeliveryAnnotations, clone.DeliveryAnnotations);
+           }
+
+           if (instance.HasSection(AmqpMessageSection.MessageAnnotations))
+           {
+               CopyDictionary(instance.MessageAnnotations, clone.MessageAnnotations);
+           }
+
+           if (instance.HasSection(AmqpMessageSection.ApplicationProperties))
+           {
+               CopyDictionary(instance.ApplicationProperties, clone.ApplicationProperties);
+           }
+
+           return clone;
+        }
+
+        /// <summary>
+        ///   Retrieves the body of an <see cref="AmqpAnnotatedMessage" /> in the form
+        ///   needed  by the <see cref="EventData.EventBody" /> accessor.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        ///
+        /// <returns>The body of the <paramref name="instance"/>, in the appropriate form for use by an <see cref="EventData" /> instance.</returns>
+        ///
+        /// <exception cref="NotSupportedException">The body of the <paramref name="instance" /> cannot be used directly by <see cref="EventData" />.</exception>
+        ///
+        public static BinaryData GetEventBody(this AmqpAnnotatedMessage instance)
+        {
+            if (instance.Body.TryGetData(out IEnumerable<ReadOnlyMemory<byte>> dataBody))
+            {
+                return BinaryData.FromBytes(MessageBody.FromReadOnlyMemorySegments(dataBody));
+            }
+
+            throw new NotSupportedException(string.Format(CultureInfo.InvariantCulture, Resources.RawAmqpBodyTypeMask, nameof(EventData), nameof(EventData.EventBody), instance.Body.BodyType, nameof(EventData.GetRawAmqpMessage)));
+        }
+
+        /// <summary>
+        ///   Retrieves the sequence number of an event from an <see cref="AmqpAnnotatedMessage" />.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        /// <param name="defaultValue">The value to return if the sequence number is not represented in the <paramref name="instance"/>.</param>
+        ///
+        /// <returns>The sequence number, if represented in the <paramref name="instance"/>; otherwise, <paramref name="defaultValue"/>.</returns>
+        ///
+        public static long GetSequenceNumber(this AmqpAnnotatedMessage instance,
+                                             long defaultValue = long.MinValue)
```

Comment:
Is the default a value one would ever use, or is it intended as a sentinel value for "this failed to return a useful value"?

Reply:
same question for the others.

Reply:
This is the common default for `EventData` to keep parity with the existing value.   I could have removed the default and force callers to provide one, since `EventData` is but this let me be a bit lazier in the tests and take advantage of existing code that assumed the event data defaults.